File: src/commands/fun/coinflip.ts
import { SlashCommandBuilder, ChatInputCommandInteraction } from 'discord.js';

export default {
  data: new SlashCommandBuilder()
    .setName('coinflip')
    .setDescription('Flip a coin'),
  async execute(interaction: ChatInputCommandInteraction): Promise<void> {
    const result = Math.random() < 0.5 ? 'Heads' : 'Tails';
    await interaction.reply({ content: `🪙 The coin landed on **${result}**.`, ephemeral: true });
  }
};

File: src/bot/commands/fun/coinflip.rs
use crate::bot::{Context, Error};

/// Flip a coin
#[poise::command(slash_command, ephemeral)]
pub async fn coinflip(ctx: Context<'_>) -> Result<(), Error> {
    let result = if rand::random::<bool>() { "Heads" } else { "Tails" };
    ctx.say(format!("🪙 The coin landed on **{result}**."))
        .await?;
    Ok(())
}
